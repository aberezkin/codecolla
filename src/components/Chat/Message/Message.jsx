import React, {Component} from 'react';
import PropTypes from "prop-types";
import './Message.styl';

class Message extends Component {
    constructor(props) {
        super(props);
    }
    render() {
        return (
            <div className="Message">
                <div className="author" style={{color: this.props.color}}>{this.props.author}</div>
                <div className="content">{this.props.content}</div>
                <div className="date">{this.props.date}</div>
            </div>
        )
    }
}

Message.propTypes = {
    author: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
};

Message.defaultProps = {
    color: 'black',
};

export default Message;