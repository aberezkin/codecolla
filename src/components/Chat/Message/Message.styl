.Message
  background-color white
  color black
  margin 2px 5px
  border 8px solid white
  border-radius 8px
  z-index 2
  .author
    font-weight bold
  .content
    font-family monospace
    white-space: pre-wrap
    word-break break-all
  .date
    font-size small
    text-align end