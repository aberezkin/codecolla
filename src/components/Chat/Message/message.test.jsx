import React from 'react';
import Message from './Message';

describe('message component', () => {
    it('should render message', () => {
        const wrapper = shallow((
            <Message
                author="me"
                content="Hello"
                date={new Date(1513954395141)}
            />
        ));

        expect(wrapper).toMatchSnapshot();
    });
});
