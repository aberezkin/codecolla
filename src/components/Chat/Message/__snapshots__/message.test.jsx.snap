// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`message component should render message 1`] = `
<div
  className="Message"
>
  <div
    className="author"
    style={
      Object {
        "color": "black",
      }
    }
  >
    me
  </div>
  <div
    className="content"
  >
    Hello
  </div>
  <div
    className="date"
  >
    '23:01:54'
  </div>
</div>
`;
